api/cloudcontroller/ccv2/service_broker_test.go

```diff
@@ -131,4 +131,160 @@ var _ = Describe("Service Broker", func() {
 			})
 		})
 	})
+
+	Describe("CreateServiceBroker", func() {
+		var (
+			serviceBroker ServiceBroker
+			warnings      Warnings
+			executeErr    error
+			brokerName    string
+			userName      string
+			password      string
+			brokerURL     string
+			spaceGUID     string
```

Comment:
We're relying on being able to set this variable in all three contexts, as if an empty spaceGUID is passed as an argument to `CreateServiceBroker`, we expect that the request that goes out to the cloud controller won't have a `space_guid` field.

Inlining this variable means we'd need to have the `JustBeforeEach` in all three contexts instead of just one; we don't think that makes the tests cleaner.

Reply:
@henryaj 
Ah good point, I think we messed that usage in the JustBeforeEach yesterday.

It's still breaking the guideline about initializing variables at the same scope as where they are declared though. It seemed like too small to block merging on, given that there isn't currently a test pollution bug with the way these contexts are written, but it would nice to have this fixed in a future PR.

The concern is that someone might add another assertion to the context where it's not being initialized and not realize that a change in test order will give different results. Or, someone else might add a new context and forget to initialize that variable and run into the same flakiness.

Reply:
Thank you very much @jenspinney and @tjvman for the review and the comments.
They really help us build better understanding around the CLI codebase and the good practices in BDD with Ginkgo.

Here is the corresponding chore in SAPI tracker to follow up on the conversation above.
https://www.pivotaltracker.com/story/show/162005881